### PR TITLE
Add credential generation info to the /osdf/namespaces endpoint

### PR DESCRIPTION
**Pull request by matyasselmeci**
[SOFTWARE-5381: Add SciToken credential generation info to Topology OSDF namespaces endpoint](https://opensciencegrid.atlassian.net/browse/SOFTWARE-5381)

### CODE REVIEW

File: src/tests/test_api.py [hunk 1]
```diff
@@ -195,6 +196,13 @@ def validate_namespace_schema(ns):
             assert isinstance(ns["usetokenonread"], bool)
             assert ns["dirlisthost"] is None or PROTOCOL_HOST_PORT_RE.match(ns["dirlisthost"])
             assert ns["writebackhost"] is None or PROTOCOL_HOST_PORT_RE.match(ns["writebackhost"])
+            credgen = ns["credential_generation"]
+            if credgen is not None:
+                assert isinstance(credgen["max_scope_depth"], int) and credgen["max_scope_depth"] > -1
+                assert credgen["strategy"] in CredentialGeneration.STRATEGIES
+                assert credgen["issuer"] and "://" in credgen["issuer"]
```
Comment (bbockelm) [inline src/tests/test_api.py:203]: Future idea: you can always parse this as a URL and ensure `netloc` isn't empty and `https` is the scheme:

```
>>> import urllib.parse
>>> urllib.parse.urlparse("https://foo.com")
ParseResult(scheme='https', netloc='foo.com', path='', params='', query='', fragment='')
```

**Review by bbockelm — APPROVED** [review]
A few ideas around URL validation improvements but it doesn't need to hold up the train.

File: src/webapp/data_federation.py [hunk 2]
```diff
@@ -99,6 +99,65 @@ def get_scitokens_conf_block(self, service_name: str):
...
+        errprefix = "CredentialGeneration:"
+
+        # Validate Strategy
+        if self.strategy not in self.STRATEGIES:
+            errors.add(f"{errprefix} invalid Strategy {self.strategy}")
+
+        # Validate Issuer
+        if not self.issuer:
+            errors.add(f"{errprefix} issuer not specified")
+        elif not isinstance(self.issuer, str) or "://" not in self.issuer:
```
[the remark this answers is not in the document]
Comment (matyasselmeci) [reply src/webapp/data_federation.py]: Done.

File: src/tests/test_api.py (hunk 1 above)
Comment (matyasselmeci) [reply src/tests/test_api.py]: Done.

**Pull request merged**